.github/workflows/mypy.yml

```diff
@@ -0,0 +1,46 @@
+name: MyPy
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - 'release/*'
+  workflow_dispatch:
+
+# Cancel previous running jobs on the same branch in case of new pushes
+concurrency:
+  group: mypy-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  mypy:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Install MyPy
+        run: |
+          pip install mypy
+          pip install ./sdk_commons ./template ./opyn ./friktion ./ribbon
+
+      - name: Run MyPy on Template
```

Comment:
Should this be a matrix?
https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#example-using-a-single-dimension-matrix

Reply:
Yep!

Reply:
It is done